### ISSUE TITLE
wx segfaulting on OSX travis tests

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#11578]
See https://travis-ci.org/matplotlib/matplotlib/jobs/400499066#L1435

Can someone with a mac reproduce this locally?

**Comment by jklymak** [comment]
With `python3.6` and `wxpython` installed via anaconda I get:

```
E               subprocess.CalledProcessError: Command '['/Users/jklymak/anaconda3/envs/matplotlibdev/bin/python', '-c', 'import sys\nfrom matplotlib import pyplot as plt, rcParams\nrcParams.update({\n    "webagg.open_in_browser": False,\n    "webagg.port_retries": 1,\n})\n\nfig = plt.figure()\nax = fig.add_subplot(111)\nax.plot([0, 1], [2, 3])\n\nif rcParams["backend"].startswith("GTK3"):\n    fig.canvas.mpl_connect("draw_event", lambda event: sys.exit(0))\nelse:\n    timer = fig.canvas.new_timer(1)\n    timer.add_callback(fig.canvas.key_press_event, "q")\n    # Trigger quitting upon draw.\n    fig.canvas.mpl_connect("draw_event", lambda event: timer.start())\n\nplt.show()\n']' returned non-zero exit status 1.
```

which is slightly different.

**Comment by anntzer** [comment]
Can you copy-paste the script (e.g. directly from test_backends_interactive) and run it (setting MPLBACKEND to wx/wxagg)?

**Comment by jklymak** [comment]
Running 

```python
import matplotlib
matplotlib.use('wxagg')
import sys
from matplotlib import pyplot as plt, rcParams
rcParams.update({
    "webagg.open_in_browser": False,
    "webagg.port_retries": 1,
})

fig = plt.figure()
ax = fig.add_subplot(111)
ax.plot([0, 1], [2, 3])

print(rcParams["backend"])

if rcParams["backend"].startswith("GTK3"):
    fig.canvas.mpl_connect("draw_event", lambda event: sys.exit(0))
else:
    timer = fig.canvas.new_timer(1)
    timer.add_callback(fig.canvas.key_press_event, "q")
    # Trigger quitting upon draw.
    fig.canvas.mpl_connect("draw_event", lambda event: timer.start())

plt.show()
```
pops up and then closes a window and prints an error to the screen:

```
pythonw testwx.py

WXAgg
/Users/jklymak/anaconda3/envs/matplotlibdev/bin/pythonw: line 3: 25249 Segmentation fault: 11  /Users/jklymak/anaconda3/envs/matplotlibdev/python.app/Contents/MacOS/python "$@"
```

**Comment by anntzer** [comment]
Can you try with older wxpython releases then?

**Comment by anntzer** [comment]
Self-merging #11593 to unbreak the build, but someone should figure out whether this needs to be fixed (or not) before 3.0 goes out!

**Comment by tacaswell** [comment]
Closing this as #11708 shows this is hanging rather than segfaulting now.
